addons/l10n_id_efaktur/tests/test_l10n_id_efaktur_download.py

```diff
@@ -0,0 +1,244 @@
+        """ Create the 2 documents on 2 separate invoices, then when we run the downlaod from the document side,
+        it should generate a zip file instead """
+        # Create 2 invoices, then generate document separately for both
+        invoice_1 = self.env["account.move"].create({
+            "move_type": "out_invoice",
+            "partner_id": self.partner_id.id,
+            "invoice_date": "2019-05-01",
+            "date": "2019-05-01",
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": self.tax_id.ids}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+        })
+        invoice_2 = self.env["account.move"].create({
+            "move_type": "out_invoice",
+            "partner_id": self.partner_id.id,
+            "invoice_date": "2019-05-01",
+            "date": "2019-05-01",
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": self.tax_id.ids}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+        })
+        invoice_1.action_post()
+        invoice_1.download_efaktur()
+        invoice_2.action_post()
+        invoice_2.download_efaktur()
+
+        # There should be total of 2 documents combined among the 2 invoices
+        invoices = invoice_1 + invoice_2
+        self.assertEqual(len(invoices.l10n_id_efaktur_document), 2)
+
+        # When downloading just the document, should generate zip file
+        action = invoices.l10n_id_efaktur_document.action_download()
+        self.authenticate(self.env.user.login, self.env.user.login)
+        result = self.url_open(url=action['url'])
+        self.assertRegex(result.headers.get('Content-Disposition'), r"efaktur.zip")
+
+    def test_efaktur_mismatch_flow_download_document(self):
+        """ Test the flow when you generate 5 invoices (inv1, .., inv5) generate document(1,2) and
+        document(3,4,5). When we try to download inv(1,2,3) redirect warning would be raised, and it should
+        provide the view of 2 documents concerned. Then, we edit such that document(1, 2) adds invoice 3.
+        Afterwards, regenerate both documents. We are expected to be able to download invoice(1,2,3) without error now,
+        while also still using the same efaktur document record as before """
+        # Generate 5 invoices
+        invoice_1 = self.env["account.move"].create({
+            "move_type": "out_invoice",
+            "partner_id": self.partner_id.id,
+            "invoice_date": "2019-05-01",
+            "date": "2019-05-01",
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": self.tax_id.ids}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+        })
+        invoice_2 = invoice_1.copy()
+        invoice_3 = invoice_1.copy()
+        invoice_4 = invoice_1.copy()
+        invoice_5 = invoice_1.copy()
```

Comment:
please create the invoices in batch.
Also if possible, limit to the lower amount of invoices possible (applies to other tests as well). That way we reduce the time it takes to run the test